sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/SpannerIO.java

```diff
@@ -227,15 +243,37 @@ public void populateDisplayData(DisplayData.Builder builder) {
     }
   }
 
+  /** Same as {@link Write} but supports grouped mutations. */
+  public static class WriteGroup extends PTransform<PCollection<MutationGroup>, PDone> {
+    private final Write spec;
+
+    public WriteGroup(Write spec) {
+      this.spec = spec;
+    }
+
+    @Override public PDone expand(PCollection<MutationGroup> input) {
+      input.apply("Write mutations to Cloud Spanner", ParDo.of(new SpannerWriteGroupFn(spec)));
+      return PDone.in(input.getPipeline());
+    }
+  }
+
+  private static class ToMutationGroupFn extends DoFn<Mutation, MutationGroup> {
+    @ProcessElement
+    public void processElement(ProcessContext c) throws Exception {
+      Mutation value = c.element();
+      c.output(MutationGroup.withPrimary(value).build());
```

Comment:
`MutationGroup` is a public class that allows to bundle several mutations together. Those are guaranteed to be submitted in a same transaction.

This is needed because a single `Mutation` is a one individual update per row, and there are use cases when a logical change is scattered across several record.

That being said, we are keeping `SpannerIO#write()` to return a `PTransform<PCollection<Mutation>, PDone>`, unless `groupped` is called. Since it is the most common use case.

Not sure if I follow your comment, it is tested in `SpannerIOTest` class. What other usage do you suggest?